### ISSUE TITLE
Add better deprecation message for `get-poetry.py` script

### ISSUE THREAD

**Issue by scratchmex** [python-poetry/poetry#6221]
When doing a new install one is tempted to use `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -` because is the first command appearing! But you should use `curl -sSL https://install.python-poetry.org | python3 -`. This is problematic because it installs under `.poetry` instead of use XDG_HOME_USER.

I have the same error following https://python-poetry.org/docs/

```
❯ curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
Retrieving Poetry metadata

This installer is deprecated. Poetry versions installed using this script will not be able to use 'self update' command to upgrade to 1.2.0a1 or later. It is recommended to use https://install.python-poetry.org instead. Instructions are available at https://python-poetry.org/docs/#installation
# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

$HOME/.poetry/bin

This path will then be added to your `PATH` environment variable by
modifying the profile files located at:

$HOME/.profile
$HOME/.zshrc

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing version: 1.1.15
  - Downloading poetry-1.1.15-linux.tar.gz (97.94MB)
```

and the error is because `This installer is deprecated. Poetry versions installed using this script will not be able to use 'self update' command to upgrade to 1.2.0a1 or later. It is recommended to use https://install.python-poetry.org instead. Instructions are available at https://python-poetry.org/docs/#installation`

If you go to the webpage it actually says `The get-poetry.py script described here will be replaced in Poetry 1.2 by install.python-poetry.org. From Poetry 1.1.7 onwards, you can already use this script as described [here](https://python-poetry.org/docs/master/#installation).`

TL;DR: use `curl -sSL https://install.python-poetry.org | python3 -`

I think this is a serious documentation error

_Originally posted by @scratchmex in https://github.com/python-poetry/poetry/issues/2148#issuecomment-1224563454_

**Comment by dimbleby** [comment]
this issue will hopefully be made redundant next week by the release of 1.2.0, and corresponding docs switch

**Comment by finswimmer** [comment]
Now that Poetry 1.2 is released and 1.1 is unmaintained we can close this.
